packages/react-sdk/src/hooks/useOptimisticDeviceStatus.ts

```diff
@@ -0,0 +1,32 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+interface Toggleable {
+  toggle: () => Promise<void>;
+}
```

Comment:
Unfortunately no way to use `InputMediaDeviceManager` type here because of private members and nominal typing.

Reply:
What would happen if `toggle()` throws an error for any reason?

Reply:
Good question. I've added handling for that case. In that case we reset the optimistic state: https://github.com/GetStream/stream-video-js/pull/1342/commits/e485ec304e616d963e09ec16515bb8c3bec0fa2f